samples/wifi/provisioning/common/wifi_prov_handler.c

```diff
@@ -0,0 +1,647 @@
+	} else {
+		prov_request_handler(&req, &rsp);
+	}
+
+	ostream = pb_ostream_from_buffer(rsp_stream.data, rsp_stream.size);
+
+	rc = pb_encode(&ostream, Response_fields, &rsp);
+	if (rc == false) {
+		/* Impossible to reply with protocol buffers. Throw error to BLE. */
+		return 0;
+	}
+	rsp_stream.len = ostream.bytes_written;
+	wifi_prov_send_rsp(&rsp_stream);
+
+	return 0;
+}
+
+/** Helper function to get the provisioning state.
+ * In case internal error occurs, it will return false.
+ */
+bool wifi_prov_is_provsioned(void)
+{
+	int rc;
+
+	rc = wifi_has_config();
+	if (rc < 0) {
+		return false;
+	}
+	return rc > 0 ? true : false;
+}
+
+/** Below part defines and registers net_mgmt event callback.
+ * These callbacks are forked in net_mgmt context, so we need
+ * to assign larger stack size to net_mgmt thread (e.g. 2048).
+ */
+/** Callback for scan result.
+ * Input: struct wifi_scan_result
+ */
+static void handle_wifi_scan_result(struct net_mgmt_event_callback *cb)
+{
+	int rc;
+	const struct wifi_scan_result *entry =
+		(const struct wifi_scan_result *)cb->info;
+	Result scan_result = Result_init_zero;
+	pb_ostream_t ostream;
+
+	NET_BUF_SIMPLE_DEFINE(scan_result_stream, 100);
+
+	scan_result.has_scan_record = true;
+	scan_result.scan_record.has_wifi = true;
+	memcpy(scan_result.scan_record.wifi.ssid.bytes, entry->ssid, entry->ssid_length);
+	scan_result.scan_record.wifi.ssid.size = entry->ssid_length;
+	memcpy(scan_result.scan_record.wifi.bssid.bytes, entry->mac,
+		entry->mac_length > 6 ? 6 : entry->mac_length);
+	scan_result.scan_record.wifi.bssid.size =
+		entry->mac_length > 6 ? 6 : entry->mac_length;
+	scan_result.scan_record.wifi.has_auth = true;
+	/* As the security is provided by device through scan request,
+	 * we just need to make sure every native security option has
+	 * one and only one mapping in AuthMode enum.
+	 * WIFI_SECURITY_TYPE_NONE = 0,		AuthMode_OPEN = 0
+	 * WIFI_SECURITY_TYPE_PSK,		AuthMode_WPA_WPA2_PSK = 4
+	 * WIFI_SECURITY_TYPE_PSK_SHA256,	AuthMode_WPA2_PSK = 3
+	 * WIFI_SECURITY_TYPE_SAE,		AuthMode_WPA3_PSK = 6
+	 * __WIFI_SECURITY_TYPE_AFTER_LAST,
+	 * WIFI_SECURITY_TYPE_MAX
+	 * WIFI_SECURITY_TYPE_UNKNOWN		AuthMode_WEP = 1,
+	 *                                      AuthMode_WPA_PSK = 2,
+	 *                                      AuthMode_WPA2_ENTERPRISE = 5
+	 */
+	if (entry->security == WIFI_SECURITY_TYPE_NONE) {
+		scan_result.scan_record.wifi.auth = AuthMode_OPEN;
+	} else if (entry->security == WIFI_SECURITY_TYPE_PSK) {
+		scan_result.scan_record.wifi.auth = AuthMode_WPA_WPA2_PSK;
+	} else if (entry->security == WIFI_SECURITY_TYPE_PSK_SHA256) {
+		scan_result.scan_record.wifi.auth = AuthMode_WPA2_PSK;
+	} else if (entry->security == WIFI_SECURITY_TYPE_SAE) {
+		scan_result.scan_record.wifi.auth = AuthMode_WPA3_PSK;
+	} else {
+		/** For other security type, use AuthMode_WEP,
+		 * as it should be rarely used now.
+		 */
+		scan_result.scan_record.wifi.auth = AuthMode_WEP;
```

Comment:
do we need to assume a default?

Reply:
It's optional field and can be absent. But I am not sure how the app will handle it. Need to confirm.

Reply:
The app will crash if auth is absent. Maybe we can keep the default value first, and allow its absence in the future, when the app can handle it properly.